core/src/test/java/org/apache/accumulo/core/iterators/system/DeletingIteratorTest.java

```diff
@@ -244,13 +245,12 @@ public void testFail() throws IOException {
     SortedKeyValueIterator<Key,Value> it =
         DeletingIterator.wrap(new SortedMapIterator(tm), false, Behavior.FAIL);
     it.seek(new Range(), EMPTY_COL_FAMS, false);
-    try {
+    assertThrows(IllegalStateException.class, () -> {
       while (it.hasTop()) {
         it.getTopKey();
         it.next();
       }
-      fail();
-    } catch (IllegalStateException e) {}
+    });
```

Comment:
This could probably be improved if it were to only assertThrows on the specific method that is being called. It's not clear here if it is expected to be throw on `hasTop`, `getTopKey`, or `next`. It's also not clear if it's supposed to be thrown on the first pass through the loop, or a subsequent one.

Reply:
Addressed in 2c61494